### DISCUSSION TITLE
Question: How to add Route Groups?

### DISCUSSION THREAD

**Discussion by ishanvyas22** [clue/framework-x#134] [Q&A]
I see this project uses `nikic/FastRoute`, it's not documented on how we can define route groups? Route groups are often used when there are multiple areas(i.e. `/admin`, `/editor`, etc.) and developers want to group assign routes to a particular prefix route.

This is already supported by the `nikic/FastRoute` package as [Route Groups](https://github.com/nikic/FastRoute#route-groups), therefore I was wondering on how to achieve this using Framework-X. I tried using `addGroup` but it doesn't seem to work ATM.

**Comment by SimonFrings** [comment]
@ishanvyas22 thanks for bringing this up :+1:

This is currently not supported in Framework X but will definitely come in the future. I'm interested, is this a must have feature for you or more like a nice to have one?

If you happen to have an idea or already an implementation, we're always happy about pull requests :100:

**Reply by ishanvyas22** [reply]
This is not a must feature for me, because I'm not working on very large project. The feature will come in handy when someone tries to build a fairly large application.

> If you happen to have an idea or already an implementation, we're always happy about pull requests

I've a few ideas about this, I'll try to make PR when I get some time.

**Comment by pfk84** [comment]
I definitely miss this feature as well, especially for API versioning like "/v1/", rate limiting of or access control/auth (via middleware) to certain endpoints groups are very handy...

**Reply by clue** [reply]
@pfk84 Thank you for your feedback and for supporting this project as a sponsor, this definitely helps us prioritize outstanding feature requests! :1st_place_medal:

I agree API versioning is a perfect use case for this, definitely plan on adding this in the future (https://github.com/clue/framework-x/discussions/134#discussioncomment-2500406) :+1:
